fix: broken link in docs/demo-scenarios.md after app/→src/ refactor

## Summary
`docs/demo-scenarios.md` references `app/hardware/README.md#dpette-8-channel-mount-replaces-so-101-gripper-jaws` but the `app/` directory was refactored to `src/` in recent commits. linkChecker fails on every PR as a result.

## Observed
```
[ERROR] <file:///.../app/hardware/README.md#dpette-8-channel-mount-replaces-so-101-gripper-jaws>
Cannot find file: File not found.
```

## Fix
Update the link in `docs/demo-scenarios.md` to the new path (likely `src/hardware/README.md#...` or the actual hardware doc location).

## Impact
Blocks CI on all Dependabot PRs (admin-merged as workaround). Fixing this unblocks green CI across the board.

--- a/docs/demo-scenarios.md
+++ b/docs/demo-scenarios.md
@@ -17,7 +17,7 @@ make setup_all      # install all dependencies + tools
 make serve          # (optional) start dashboard on :8080
 ```
 
-**Physical demo setup (optional).** The pipetting use cases (UC1, UC5) can be demonstrated on real hardware once the printed parts and pipette are mounted. The DLAB dPette+ 8-channel mount is the primary configuration — see [../app/hardware/README.md](../app/hardware/README.md#dpette-8-channel-mount-replaces-so-101-gripper-jaws) for the assembly and print settings, [../docs/hardware/BOM.md](hardware/BOM.md) for the parts list, and [prusa-mk4-ops.md](hardware/prusa-mk4-ops.md) for uploading gcode to a Prusa MK4 via PrusaLink. All demo commands below also run cleanly in **stub mode** without any hardware.
+**Physical demo setup (optional).** The pipetting use cases (UC1, UC5) can be demonstrated on real hardware once the printed parts and pipette are mounted. The DLAB dPette+ 8-channel mount is the primary configuration — see [../src/hardware/README.md](../src/hardware/README.md#dpette-8-channel-mount-replaces-so-101-gripper-jaws) for the assembly and print settings, [../docs/hardware/BOM.md](hardware/BOM.md) for the parts list, and [prusa-mk4-ops.md](hardware/prusa-mk4-ops.md) for uploading gcode to a Prusa MK4 via PrusaLink. All demo commands below also run cleanly in **stub mode** without any hardware.
 
 ## UC1: Pipette a 96-Well Plate
 

